### PR TITLE
remove thin USD wrappers from C++ API in favor of underlying USD functions

### LINKED ISSUE #390: Remove functions from C++ API
There are several functions in the C++ API that are thin wrappers around adding or removing prims from the USD stage. We should remove these functions from `CesiumOmniverse.h/cpp` and `Context.h/cpp` and use the Python USD API directly.

These functions can be removed to start:

* `addTilesetUrl`
* `addTilesetIon`
* `addImageryIon`
* `addTilesetAndImagery`
* `isTileset`
* `removeTileset`

For reference, look at how `cesium.performance.app` adds and removes tilesets without touching the C++ API.

### PR DESCRIPTION
**Pull request by mattelser**
fixes #390

The only thing that calls remove_tileset is clicking the `remove all tilesets` button of the debug window

### CODE REVIEW

**Comment by lilleyse** [comment]
I'm still going through my review, but as an early comment these two areas need to be updated:

* https://github.com/CesiumGS/cesium-omniverse/blob/252d8cf2c017fcd30245ea370d3da477837b3d54/exts/cesium.omniverse/cesium/omniverse/extension.py#L313-L315
* https://github.com/CesiumGS/cesium-omniverse/blob/993e8f2bb983a991994908b0c6eaa7e977c0967f/exts/cesium.omniverse/cesium/omniverse/ui/asset_details_widget.py#L78

File: exts/cesium.omniverse/cesium/omniverse/utils/utils.py
Comment (lilleyse) [inline exts/cesium.omniverse/cesium/omniverse/utils/utils.py:1]: Style notes throughout the code:

* use camel_case for variable names
* add return type annotations

File: exts/cesium.omniverse/cesium/omniverse/utils/utils.py [hunk 1]
```diff
@@ -34,3 +42,50 @@ def str_is_empty_or_none(s: Optional[str]):
...
+    parent = imagery.GetPrim().GetParent()
+    assert parent.IsA(CesiumTileset)
+
+    imagery.GetIonAssetIdAttr().Set(asset_id)
+    imagery.GetIonAccessTokenAttr().Set(token)
+
+    return imagery_path
+
+
+def is_tileset(maybeTileset):
```
Comment (lilleyse) [inline exts/cesium.omniverse/cesium/omniverse/utils/utils.py:84]: `maybeTileset` should have a type annotation.

File: exts/cesium.omniverse/cesium/omniverse/utils/utils.py
Comment (lilleyse) [inline exts/cesium.omniverse/cesium/omniverse/utils/utils.py:1]: For consistency and simplicity, I think both arguments and return variables for the USD util functions should be `str` types.

Comment (lilleyse) [inline exts/cesium.omniverse/cesium/omniverse/utils/utils.py:1]: Could you write tests for the new functions in here?

**Comment by lilleyse** [comment]
Other things:

* CI is failing
* Python unit tests are failing (though this may be happening on `main` as well)
* Could you merge `main` into this branch?

File: exts/cesium.omniverse/cesium/omniverse/utils/utils.py [hunk 2]
```diff
@@ -34,3 +42,50 @@ def str_is_empty_or_none(s: Optional[str]):
         return True
 
     return False
+
+
+def add_tileset_ion(name: str, tilesetId: int, token: str = ""):
```
Comment (weegeekps) [inline exts/cesium.omniverse/cesium/omniverse/utils/utils.py:47]: None of the tileset stuff belongs in this file. It should be moved to a different file. The utils file is for small UI related utilities.

File: exts/cesium.omniverse/cesium/omniverse/utils/utils.py
Comment (mattelser) [reply exts/cesium.omniverse/cesium/omniverse/utils/utils.py:1]: agreed those would be a good idea, but probably out of scope for this PR. I've suggested them in #310

File: exts/cesium.omniverse/cesium/omniverse/utils/utils.py (hunk 2 above)
Comment (mattelser) [reply exts/cesium.omniverse/cesium/omniverse/utils/utils.py]: fair point, I've moved these into a `usdUtils` module

File: exts/cesium.usd.plugins/cesium/usd/plugins/CesiumUsdSchemas/__init__.pyi [hunk 3]
```diff
@@ -146,7 +146,7 @@ class Session(pxr.Usd.Typed):
     @classmethod
     def __reduce__(cls) -> Any: ...
 
-class Tileset(pxr.UsdGeom.Boundable):
+class Tileset(pxr.UsdGeom.Gprim):
```
Comment (lilleyse) [inline exts/cesium.usd.plugins/cesium/usd/plugins/CesiumUsdSchemas/__init__.pyi:149]: Thanks for fixing this 👍

**Review by lilleyse — APPROVED** [review]

File: exts/cesium.omniverse/cesium/omniverse/utils/utils.py (hunk 2 above)
Comment (weegeekps) [reply exts/cesium.omniverse/cesium/omniverse/utils/utils.py]: Thanks for fixing this.

**Review by weegeekps — APPROVED** [review]

**Pull request merged**